### ISSUE TITLE
Add ARM support

### ISSUE THREAD

**Issue by PsychicNoodles** [CleverCloud/kawa#4]
Currently, x86 arch specific instructions are used in a couple utility functions, [here](https://github.com/CleverCloud/kawa/blob/68c22f4e1df35b8855b947c1499b05912aec6bca/src/protocol/utils.rs#L123) and [here](https://github.com/CleverCloud/kawa/blob/68c22f4e1df35b8855b947c1499b05912aec6bca/src/protocol/utils.rs#L179). I'm trying to run sozu on a Raspberry Pi, so it would be nice to have a version that can build and run there, even if it ends up not benefiting from simd optimizations.

**Comment by 5310** [comment]
Came to second this. On a branch new Ampere VPS. This is the only thing so far that won't work on ARM.

**Comment by Wonshtrum** [comment]
I'm sorry, I was in a rush when I merged the "perf" branch and I knew SIMD would cause problems.
I will soon contribute again to Kawa (mainly to add the missing h2 parts) and I will put all the SIMD stuff under a feature flag (like I should have done). Sozu will get a corresponding feature to enable or disable SIMD in Kawa.

**Comment by Wonshtrum** [comment]
SIMD has been moved under a feature flag. It will be tested and ported to Sozu soon.

**Comment by Wonshtrum** [comment]
I apologize for the delay, the Sozu team and I had to prioritize the development of HTTP2, which led to a longer wait than expected.
However, I'm pleased to announce that we've now released kawa 0.6.3. This release includes important bug fixes and introduces the SIMD feature flag.
Shortly after that, we also released Sozu 0.15.4, which exposes the SIMD feature flag.
I hope Sozu compiles properly on ARM when this feature is disabled (it is enabled by default).

**Comment by lu-zero** [comment]
maybe would be a good idea to guard it using the right cfg entries instead of a feature?

**Comment by Wonshtrum** [comment]
Yes, that's also our goal. I don't like the way parsing is handled presently. The fast parsers and simd were an afterthought and it shows.
Our efforts are currently on the http2 support, which is much more complex than we anticipated. Kawa might be rewritten altogether when we are sure to understand exactly how we want it to work.
